File: 50-powx-n/powx-n.cpp
class Solution {
public:
    double power(double n, int x){
    if(x == 0){
        return 1;
    }
    double half = power(n,x/2);
    if(x % 2 == 0){
        return half * half;
    }
    else{
        return half * half * n;
    }
}
double change(double n ,int x){
    long long a = x;
    if(x < 0){
        
        return 1.0/power(n,a);
    }
    return power(n,a);
}
    double myPow(double x, int n) {
        if(x==1){
            return 1;
        }
        return change(x,n);
    }
};

File: 50-powx-n/powx-n.java
class Solution {
    public double ans(double x , int n){
        if (n == 0) {
            return 1;
        }

        double ans1 = ans(x, n / 2);

        if (n % 2 == 0) {
            return ans1 * ans1;
        } else {
            return x * ans1 * ans1;
        }
    }


    public double myPow(double x, int n) {
    

        if(n < 0){
            return 1.0/ ans(x,-n);
        }

        return ans(x,n);        

    }
}